### PR TITLE
Fix stock-detail watchlist membership state

### LINKED ISSUE #353: Stock detail watchlist menu ignores existing membership and silently swallows duplicate-add errors
## Summary

On the stock detail page, the **Add to Watchlist** menu does not reflect whether the current stock is already present in a watchlist.

If the stock is already in the selected watchlist, the UI still allows the user to select that watchlist and sends a duplicate-add request.

The backend correctly rejects the duplicate, but the frontend provides no visible feedback, so from the user's perspective nothing happens.

## Example

Stock: `MSFT`

`MSFT` is already present in the `Portfolio` watchlist.

From:

`/stocks/MSFT`

Open:

`Add to Watchlist -> Portfolio`

The frontend sends:

```http
POST /api/v1/user-watchlists/1/items
```

Payload:

```json
{
  "symbol": "MSFT"
}
```

Backend response:

```http
400 Bad Request
```

Response body:

```json
{
  "detail": "Stock already exists in watchlist"
}
```

The UI displays no error or status change.

## Current Behavior

The dropdown still shows `Portfolio` as an available add target even though `MSFT` is already a member.

Example:

```text
Add MSFT to:

Portfolio
media-to-analyze
Create new watchlist
```

Selecting `Portfolio` sends the POST request, receives the duplicate error, and then silently does nothing.

## Expected Behavior

The stock detail page should be aware of the stock's existing watchlist memberships.

For example:

```text
Add MSFT to:

✓ Portfolio            Already added
  media-to-analyze
+ Create new watchlist
```

A watchlist that already contains the current ticker should either:

* be disabled and marked as already added, or
* provide an appropriate remove/toggle action.

The frontend should also display an error/toast for failed API requests instead of silently swallowing them.

For this specific duplicate case, a message such as:

```text
MSFT is already in Portfolio
```

would be sufficient.

## Backend

Duplicate protection appears to be working correctly.

The problem does not appear to be a watchlist size/capacity limit.

The backend currently returns:

```http
400 Bad Request
```

with:

```json
{
  "detail": "Stock already exists in watchlist"
}
```

Optionally, `409 Conflict` may be a more semantically appropriate status for duplicate membership, but this is secondary to the frontend issue.

## Requirements

* [ ] Determine the current ticker's membership in each user watchlist before rendering the menu.
* [ ] Mark watchlists that already contain the ticker as already added.
* [ ] Prevent duplicate add requests from the UI.
* [ ] Refresh membership state after successful add/remove operations.
* [ ] Surface API failures to the user instead of silently ignoring them.
* [ ] Preserve backend duplicate protection.
* [ ] Add tests covering duplicate membership and stock-detail watchlist state.

## Acceptance Criteria

Given a stock that is already in `Portfolio`:

1. Opening **Add to Watchlist** shows `Portfolio` as already containing the stock.
2. The UI does not send another add request for that same stock/watchlist pair.
3. Other watchlists that do not contain the stock remain selectable.
4. Adding the stock to another watchlist updates the menu state immediately.
5. Any unexpected API error is shown visibly to the user.

### PR DESCRIPTION
**Pull request by xang1234**
## Summary
- add a lightweight API for resolving symbol membership across user watchlists
- mark existing memberships as already added and prevent duplicate requests
- refresh membership whenever the menu opens and block actions while that refresh is in flight
- surface watchlist API failures in the menu and update membership immediately after successful adds
- add backend and frontend regression coverage

## Testing
- `cd backend && ./venv/bin/pytest tests/unit/test_stock_workspace_endpoints.py -q` (17 passed)
- `cd frontend && npm run test:run` (671 passed)
- `cd frontend && npm run build`
- `cd frontend && npm run lint` (0 errors; 4 existing warnings)

Closes #353

## Summary by CodeRabbit

* **New Features**
  * Added watchlist membership lookup for multiple stock symbols.
  * Watchlist menus identify symbols already added and disable duplicate additions.
  * Membership information refreshes when the menu is reopened.
  * Symbols are normalized and deduplicated for consistent watchlist actions.

* **Bug Fixes**
  * Prevented duplicate watchlist additions.
  * Improved handling of partial bulk additions and invalid symbol formats.
  * Added clear error messages when watchlist actions fail.

### CODE REVIEW

File: frontend/src/components/common/AddToWatchlistMenu.jsx [hunk 1]
```diff
@@ -24,37 +24,71 @@ import {
...
   const [newWatchlistName, setNewWatchlistName] = useState('');
   const [showNewInput, setShowNewInput] = useState(false);
-  const [addedToIds, setAddedToIds] = useState(new Set());
   const queryClient = useQueryClient();
 
   const open = Boolean(anchorEl);
 
   // Normalize symbols to array
   const symbolList = Array.isArray(symbols) ? symbols : [symbols];
+  const normalizedSymbols = [...new Set(symbolList.map((symbol) => symbol.toUpperCase()))];
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/common/AddToWatchlistMenu.jsx]: Implemented in 609a6dfb. The component now trims symbols, strips leading dollar signs, uppercases, and deduplicates once, then reuses that canonical list for membership lookups and both single/bulk add payloads. A regression test covers equivalent inputs such as ` $msft ` and `MSFT`.

File: frontend/src/components/common/AddToWatchlistMenu.test.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  it('refreshes membership whenever the menu reopens despite the app cache window', async () => {
+    const queryClient = new QueryClient({
+      defaultOptions: {
+        queries: { retry: false, staleTime: 300_000 },
+      },
+    });
+    api.getWatchlistMemberships.mockResolvedValueOnce({
+      memberships: { MSFT: [] },
+    });
+    render(
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/common/AddToWatchlistMenu.test.jsx]: Implemented in 609a6dfb. `renderWithProviders` now accepts an optional injected QueryClient, and the stale-cache test uses the shared renderer instead of constructing QueryClientProvider directly.

File: frontend/src/components/common/AddToWatchlistMenu.jsx [hunk 3]
```diff
@@ -83,15 +119,18 @@ function AddToWatchlistMenu({ symbols, trigger, onSuccess, size = 'small' }) {
     mutationFn: ({ watchlistId, symbols }) => bulkAddItems(watchlistId, symbols),
     onSuccess: (_, variables) => {
       queryClient.invalidateQueries({ queryKey: ['userWatchlistData', variables.watchlistId] });
-      setAddedToIds((prev) => new Set([...prev, variables.watchlistId]));
+      recordMembership(variables.watchlistId);
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/common/AddToWatchlistMenu.jsx]: Implemented in 7ac640ff. Bulk-add success now updates the membership cache only for symbols present in the endpoint response, so skipped or unknown symbols remain unrecorded and the watchlist stays selectable. Added a regression test for a partial-success response; the full frontend suite passes with 673 tests.

File: frontend/src/components/common/AddToWatchlistMenu.jsx [hunk 4]
```diff
@@ -73,25 +109,31 @@ function AddToWatchlistMenu({ symbols, trigger, onSuccess, size = 'small' }) {
     mutationFn: ({ watchlistId, data }) => addItem(watchlistId, data),
     onSuccess: (_, variables) => {
       queryClient.invalidateQueries({ queryKey: ['userWatchlistData', variables.watchlistId] });
-      setAddedToIds((prev) => new Set([...prev, variables.watchlistId]));
+      recordMembership(variables.watchlistId);
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/common/AddToWatchlistMenu.jsx]: Implemented in 13d9cb3f. Membership cache writes now derive both the query key and recorded symbol from the submitted mutation variables, so a prop change during an in-flight add cannot mark the newly displayed symbol. Added a deferred-mutation regression test that rerenders from MSFT to AAPL before completion.

File: frontend/src/components/common/AddToWatchlistMenu.jsx [hunk 5]
```diff
@@ -24,37 +24,73 @@ import {
...
   getWatchlists,
+  getWatchlistMemberships,
   createWatchlist,
   addItem,
   bulkAddItems,
 } from '../../api/userWatchlists';
 
+const getErrorMessage = (error) => (
+  error?.response?.data?.detail || error?.message || 'Unable to update watchlist'
+);
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/common/AddToWatchlistMenu.jsx:55]: Implemented in 13d9cb3f. Error details are now normalized into displayable text: strings pass through, arrays are joined, object `msg` fields are preferred, and other objects are serialized. A regression test covers FastAPI’s standard structured validation detail.
